chapters/publication.tex

```diff
@@ -99,91 +181,265 @@ \section{Collaborating on academic writing}
+so that readers can understand which data components are
+collected directly in the field and which are derived.
+You should provide a clean version of the data
+which corresponds exactly to the original database or questionnaire
+as well as the constructed or derived dataset used for analysis.
+Wherever possible, you should also release the code
+that constructs any derived measures,
+particularly where definitions may vary,
+so that others can learn from your work and adapt it as they like.
+
+\subsection{Publishing code for replication}
+
+Before publishing your code, you should edit it for content and clarity
+just as if it were written material.
+The purpose of releasing code is to allow others to understand
+exactly what you have done in order to obtain your results,
+as well as to apply similar methods in future projects.
+Therefore it should both be functional and readable.
+Code is often not written this way when it is first prepared,
+so it is important for you to review the content and organization
+so that a new reader can figure out what and how your code should do.
+Therefore, whereas your data should already be very clean at this stage,
+your code is much less likely to be so, and this is where you need to make
+time investments prior to releasing your replication package.
+By contrast, replication code usually has few legal and privacy constraints.
+In most cases code will not contain identifying information;
+but make sure to check carefully that it does not.
+Publishing code also requires assigning a license to it;
+in a majority of cases, code publishers like GitHub
+offer extremely permissive licensing options by default.
+(If you do not provide a license, nobody can use your code!)
+
+Make sure the code functions identically on a fresh install of your chosen software.
+A new user should have no problem getting the code to execute perfectly.
+In either a scripts folder or in the root directory,
+include a master script (dofile or R script for example).
+The master script should allow the reviewer
+to change a single line of code: the one setting the directory path.
+After that, running the master script should run the entire project
+and re-create all the raw outputs exactly as supplied.
+Indicate the filename and line to change.
+Check that all your code will run completely on a new computer:
+Install any required user-written commands in the master script
+(for example, in Stata using \texttt{ssc install} or \texttt{net install}
+and in R include code giving users the option to install packages,
+including selecting a specific version of the package if necessary).
+In many cases you can even directly provide the underlying code
+for any user-installed packages that are needed to ensure forward-compatibility.
+Make sure system settings like \texttt{version}, \texttt{matsize}, and \texttt{varabbrev} are set.
+
+Finally, make sure that the code and its inputs and outputs are clearly identified.
```

Comment:
link this back to the section on master scripts in chapter 6

Reply:
519f9de65d2fd09e94b6cc1d1d71c5fe97bde34a